### PR TITLE
coot/cardano ping

### PR DESCRIPTION
**Pull request by coot**
- Add support for V7 keepalive
- cardano-ping: adhare to keep alive protocol

### CODE REVIEW

File: network-mux/demo/cardano-ping.hs [hunk 1]
```diff
@@ -222,6 +222,12 @@ keepAliveReqEnc _ cookie =
 keepAliveReq :: NodeVersion -> Word16 -> ByteString
 keepAliveReq v c = CBOR.toLazyByteString $ keepAliveReqEnc v c
 
+keepAliveDone :: ByteString
+keepAliveDone =
+    CBOR.toLazyByteString $
+         CBOR.encodeListLen 1
+      <> CBOR.encodeWord 2
```
Comment (karknu) [inline network-mux/demo/cardano-ping.hs:229]: You should cover the pre v7 version of the protocol too.

**Review by karknu — CHANGES_REQUESTED** [review]

File: network-mux/demo/cardano-ping.hs (hunk 1 above)
Comment (coot) [reply network-mux/demo/cardano-ping.hs:229]: Good point

Comment (coot) [reply network-mux/demo/cardano-ping.hs]: Note the shortcoming of the approach of integration of version and version data: we need to pattern match on the version number: when we will add `v8` for `node-to-node` the `cardano-ping` will break if one forgets to update the encoder.

Comment (coot) [reply network-mux/demo/cardano-ping.hs]: I pushed one more commit to fix this.

**Review by karknu — APPROVED** [review]
LGTM

**Comment by coot** [comment]
bors merge

**Pull request merged**